### ISSUE TITLE
Migration rename not working

### ISSUE THREAD

**Issue by Narquadah** [typeorm/typeorm#393]
While evaluating typeorm for a project I came across a problem with migrations. I tried it with sqlite and postgres.

Using this migration:

```ts
    public async up(queryRunner: QueryRunner, connection: Connection, entityManager?: EntityManager): Promise<any> {
        await queryRunner.renameColumn('article', 'ttle', "title")
    }

    public async down(queryRunner: QueryRunner, connection: Connection, entityManager?: EntityManager): Promise<any> {
        await queryRunner.renameColumn('article', "title", 'ttle')
    }

```

First with sqlite:

```
executing query: SELECT * FROM sqlite_master WHERE type = 'table' AND name = 'migrations'
executing query: SELECT * FROM "migrations" "migrations"
0 migrations are already loaded in the database.
1 migrations were found in the source code.
1 migrations are new migrations that needs to be executed.
executing query: BEGIN TRANSACTION
executing query: SELECT * FROM sqlite_master WHERE type = 'table' AND name IN ('article')
executing query: PRAGMA table_info("article")
executing query: PRAGMA index_list("article")
executing query: PRAGMA foreign_key_list("article")
executing query: CREATE TABLE "temporary_article" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "author" character varying(300) NOT NULL, "likes" integer NOT NULL, "text" text NOT NULL, "ttle" text NOT NULL, "created" date NOT NULL)
executing query: INSERT INTO "temporary_article"("id", "author", "likes", "text", "ttle", "created") SELECT "id", "author", "likes", "text", "ttle", "created" FROM "article"
executing query: DROP TABLE "article"
executing query: ALTER TABLE "temporary_article" RENAME TO "article"
executing query: INSERT INTO "migrations"("timestamp", "name") VALUES ($1,$2) -- PARAMETERS: [1491815623631,"FixFirstIteration1491815623631"]
Migration FixFirstIteration1491815623631 has been executed successfully.
executing query: COMMIT
```
While it works without errors nothing happened. 

And then with postgres:

```sh
executing query: SELECT * FROM information_schema.tables WHERE table_catalog = 'test' AND table_schema = 'public' AND table_name = 'migrations'
executing query: SELECT * FROM "migrations" "migrations"
0 migrations are already loaded in the database.
1 migrations were found in the source code.
1 migrations are new migrations that needs to be executed.
executing query: START TRANSACTION
executing query: SELECT * FROM information_schema.tables WHERE table_catalog = 'test' AND table_schema = 'public' AND table_name IN ('article')
executing query: SELECT * FROM information_schema.columns WHERE table_catalog = 'test' AND table_schema = 'public'
executing query: SELECT t.relname AS table_name, i.relname AS index_name, a.attname AS column_name  FROM pg_class t, pg_class i, pg_index ix, pg_attribute a
WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid
AND a.attnum = ANY(ix.indkey) AND t.relkind = 'r' AND t.relname IN ('article') ORDER BY t.relname, i.relname
executing query: SELECT table_name, constraint_name FROM information_schema.table_constraints WHERE table_catalog = 'test' AND constraint_type = 'FOREIGN KEY'
executing query: SELECT * FROM information_schema.table_constraints WHERE table_catalog = 'test' AND constraint_type = 'UNIQUE'
executing query: SELECT c.column_name, tc.table_name, tc.constraint_name FROM information_schema.table_constraints tc
JOIN information_schema.constraint_column_usage AS ccu USING (constraint_schema, constraint_name)
JOIN information_schema.columns AS c ON c.table_schema = tc.constraint_schema AND tc.table_name = c.table_name AND ccu.column_name = c.column_name
where constraint_type = 'PRIMARY KEY' and tc.table_catalog = 'test'
executing query: ALTER TABLE "article" ALTER COLUMN "ttle" RENAME TO title
query failed: ALTER TABLE "article" ALTER COLUMN "ttle" RENAME TO title
error during executing query:error: syntax error at or near "RENAME"
```
In postgres the was the basic query `ALTER TABLE "article" ALTER COLUMN "ttle" RENAME TO title` wrong, it should be `ALTER TABLE "article" RENAME ttle TO title`

Any help would be appreciated.

**Comment by pleerock** [comment]
Looks like a bug. Use manually `queryRunner.query(\`ALTER TABLE "article" RENAME "ttle" to "title"\`)` query instead.

**Comment by aluco100** [comment]
Hi I'm trying to do the same thing but for sqlite It can't rename columns so I do the following migration:

```
export class ProductRefactoring1515695789696 implements MigrationInterface {

    public async up(queryRunner: QueryRunner): Promise<any> {
      await queryRunner.query("ALTER TABLE product RENAME TO tmp_product;");
      await queryRunner.query(`CREATE TABLE IF NOT EXISTS product (
                        id INTEGER PRIMARY KEY AUTOINCREMENT,
                        title VARCHAR(45),
                        timestamp INT,
                        description MEDIUMTEXT,
                        quantity INT,
                        photo VARCHAR(45),
                        status INT)`);
      await queryRunner.query(`INSERT INTO product(id, title,timestamp,description,quantity,photo,status)
                              SELECT id, name,timestamp,description,quantity,photo,status FROM tmp_product;`);
      await queryRunner.query("DROP TABLE tmp_product;");
    }

    public async down(queryRunner: QueryRunner): Promise<any> {
      await queryRunner.query("ALTER TABLE product RENAME TO tmp_product;");
      await queryRunner.query(`CREATE TABLE IF NOT EXISTS product (
                        id INTEGER PRIMARY KEY AUTOINCREMENT,
                        name VARCHAR(45),
                        timestamp INT,
                        description MEDIUMTEXT,
                        quantity INT,
                        photo VARCHAR(45),
                        status INT)`);
      await queryRunner.query(`INSERT INTO product(id, name,timestamp,description,quantity,photo,status)
                              SELECT id, title,timestamp,description,quantity,photo,status FROM tmp_product;`);
      await queryRunner.query("DROP TABLE tmp_product;");
    }

}

```

Why the method `renameColumn` is not working?

Best regards

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.

**Comment by ChumachenkoMihail** [comment]
> Hi I'm trying to do the same thing but for sqlite It can't rename columns so I do the following migration:
> 
> ```
> export class ProductRefactoring1515695789696 implements MigrationInterface {
> 
>     public async up(queryRunner: QueryRunner): Promise<any> {
>       await queryRunner.query("ALTER TABLE product RENAME TO tmp_product;");
>       await queryRunner.query(`CREATE TABLE IF NOT EXISTS product (
>                         id INTEGER PRIMARY KEY AUTOINCREMENT,
>                         title VARCHAR(45),
>                         timestamp INT,
>                         description MEDIUMTEXT,
>                         quantity INT,
>                         photo VARCHAR(45),
>                         status INT)`);
>       await queryRunner.query(`INSERT INTO product(id, title,timestamp,description,quantity,photo,status)
>                               SELECT id, name,timestamp,description,quantity,photo,status FROM tmp_product;`);
>       await queryRunner.query("DROP TABLE tmp_product;");
>     }
> 
>     public async down(queryRunner: QueryRunner): Promise<any> {
>       await queryRunner.query("ALTER TABLE product RENAME TO tmp_product;");
>       await queryRunner.query(`CREATE TABLE IF NOT EXISTS product (
>                         id INTEGER PRIMARY KEY AUTOINCREMENT,
>                         name VARCHAR(45),
>                         timestamp INT,
>                         description MEDIUMTEXT,
>                         quantity INT,
>                         photo VARCHAR(45),
>                         status INT)`);
>       await queryRunner.query(`INSERT INTO product(id, name,timestamp,description,quantity,photo,status)
>                               SELECT id, title,timestamp,description,quantity,photo,status FROM tmp_product;`);
>       await queryRunner.query("DROP TABLE tmp_product;");
>     }
> 
> }
> ```
> 
> Why the method `renameColumn` is not working?
> 
> Best regards

It happens bcz u use entities. U just need to change column name in entity
